src/signalr-service.ts

```diff
@@ -0,0 +1,53 @@
+import { Subject } from 'rxjs';
+import { HubConnection, HubConnectionBuilder, LogLevel, HubConnectionState } from '@aspnet/signalr';
+import { AzureWebCommandService } from './azure-web-command-service';
+import { Guid } from 'guid-typescript';
+import { SignalRResponse } from './signal-r-response';
+
+export class SignalRService {
+    private signalrMessage: Subject<any> = new Subject<any>();
+    private hubConnection: HubConnection;
+
+    /**
+     * Create a new instance of SignalRService
+     * @param signalrUrl The azure signalR endpoint.
+     * @param enableLogging Enable logging with level "Debug".
+     */
+    constructor(signalrUrl: string, enableLogging?: boolean) {
+        const connection = new HubConnectionBuilder()
+        .withUrl(signalrUrl);
+        if (enableLogging) {
+            connection.configureLogging(LogLevel.Debug);
+        }
+
+        this.hubConnection = connection.build();
+        this.hubConnection.start()
```

Comment:
I'm a bit sceptical on doing a lot of logic in the constructor. Init functions are a better fit I think. Also makes unit testing easier.

Reply:
changed